### PR TITLE
Refactor: Simplify Article Follow buttons

### PR DESCRIPTION
**Pull request by eessex**
- Remove non-publishing logic from follow button components
- Use cohesion follow/unfollow events
- Adds `AnalyticsContext`
- Remove `onOpenAuthModal` prop from articles in favor of calling mediator via `SystemContext`

Required by https://github.com/artsy/force/pull/6375
https://artsyproduct.atlassian.net/browse/PLATFORM-2775

<details>
  <summary>📦 Published PR as canary version: <code>28.7.1-canary.3627.63333.0</code></summary>
  <br />
  
  :sparkles: Test out this PR locally via:
  
  ```sh
  npm install @artsy/reaction@28.7.1-canary.3627.63333.0
  # or 
  yarn add @artsy/reaction@28.7.1-canary.3627.63333.0
  ```
</details>

### CODE REVIEW

File: src/Components/FollowButton/FollowButton.tsx [hunk 1]
```diff
@@ -40,35 +38,36 @@ export class FollowButtonDeprecated extends React.Component<Props, State> {
...
+        <Sans size="2" weight="medium">
+          {text}
+        </Sans>
       </FollowButtonContainer>
     )
   }
 }
 
-interface DivProps {
+const FollowButtonContainer = styled(Clickable)<{
```
Comment (dzucconi) [inline src/Components/FollowButton/FollowButton.tsx:49]: ❤️

File: src/Components/FollowButton/FollowButton.tsx [hunk 2]
```diff
@@ -40,35 +38,36 @@ export class FollowButtonDeprecated extends React.Component<Props, State> {
         onMouseEnter={() => this.setState({ showUnfollow: true })}
         onMouseLeave={() => this.setState({ showUnfollow: false })}
       >
-        {text}
+        <Sans size="2" weight="medium">
```
Comment (dzucconi) [inline src/Components/FollowButton/FollowButton.tsx:41]: Consider [the updated type treatments](https://palette.artsy.net/tokens/typography/)? — `<Text variant="mediumText">`

**Review by dzucconi — APPROVED** [review]

File: src/Components/FollowButton/FollowButton.tsx (hunk 2 above)
Comment (eessex) [reply src/Components/FollowButton/FollowButton.tsx:41]: Ya this should change - switching this to use the `Button` instead, which is what we're doing in Force
Screen Shot 2020-09-30 at 2 48 20 PM

In general though, lots of old text, text styles that are pre-palette in Reaction -- articles could definitely use a typography update.

**Comment by artsyit** [comment]
:rocket: PR was released in [`v28.8.0`](https://github.com/artsy/reaction/releases/tag/v28.8.0) :rocket:

**Pull request merged**